Recommendations

DesidiamS opened the pull request.

Commit:
Реализован третий этап дипломного проекта

Commit:
Почистил мусорный код

Commit:
Изменил родительский модуль

Commit:
Исправлен pom у serialization

Commit:
Исправлен pom у stats-client

Commit:
Убран не нужный модуль stats-server

Commit:
Убран не нужный модуль stats-server

Commit:
Добавлен билд для analyzer

Commit:
Добавлена настройка для клиента grpc analyzer

Commit:
Добавлена зависимость для того, чтобы сервисы обнаруживались в discovery

Commit:
Изменены package для analyzer

Commit:
Изменена работа analyzer

Commit:
Доработан сервис взаимодействия пользователя

bismuthD commented on stats/analyzer/src/main/java/ru/practicum/grpc/AnalyzerController.java:
```diff
@@ -0,0 +1,56 @@
+            iterator.forEachRemaining(responseObserver::onNext);
+            responseObserver.onCompleted();
+        } catch (Exception e) {
+            responseObserver.onError(e);
```
Здесь стоит добавить логирование

bismuthD commented on stats/analyzer/src/main/java/ru/practicum/service/InteractionService.java:
```diff
@@ -0,0 +1,53 @@
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class InteractionService {
```
@Transactional(readOnly = true)

bismuthD commented on core/event/src/main/java/ru/practicum/event/controller/PublicEventController.java:
```diff
@@ -85,25 +82,28 @@ public ResponseEntity<List<EventShortDto>> getEventsByFilters(@RequestParam(name
 
+    @PutMapping("/{eventId}")
+    public void putLike(@PathVariable Long eventId,
+                        @RequestHeader("X-EWM-USER-ID") long userId) {
```
значение X-EWM-USER-ID лучше вынести в константу на уровне класса

bismuthD reviewed (commented):
В целом все отлично, оставил несколько комментариев по улучшению

Commit:
Добавлено логирование в analyzer
Добавлена аннотация Transactional(readOnly)
Значение хэдера для пользователя вынесена в константу класса
